lib/helpers/prefix-event-handler.js

```diff
@@ -0,0 +1,45 @@
+module.exports = prefixEventHandler
+
+function prefixEventHandler (prefix, prefixState) {
+  // only listen to events from parent if there is a handler
+  prefixState.emitter.on('removeListener', function (type, handler) {
+    updateListening(false, prefixState, prefixHandler)
+  })
+  prefixState.emitter.on('newListener', function (type, handler) {
+    updateListening(true, prefixState, prefixHandler)
+  })
+
+  function prefixHandler (eventName, object) {
+    if (object._id.substr(0, prefix.length) !== prefix) {
+      return
+    }
+
+    prefixState.emitter.emit('change', eventName, object)
+    prefixState.emitter.emit(eventName, object)
+  }
+}
+
+function updateListening (wasAdded, prefixState, handler) {
+  // newListener event will be emitted before the listener will be added!
+  // But the removeListener event after the listener was removed!
+  // That is why wasAdded is needed.
+  var shouldListen = wasAdded || [
```

Comment:
I properly should pass the eventName if it was added, and check it in the .some-function `count > 0 || type === eventName`.
This would make sure, that if a unknown event type was added, it wouldn't count.